pkg/hubble/metrics/metrics.go

```diff
@@ -146,3 +136,35 @@ func InitMetrics(reg *prometheus.Registry, enabled api.Map, grpcMetrics *grpc_pr
 
 	return nil
 }
+
+func EnableMetrics(log logrus.FieldLogger, metricsTLSConfig *certloader.WatchedServerConfig, grpcMetrics *grpc_prometheus.ServerMetrics, enableOpenMetrics bool) {
+	srv := http.Server{
+		Addr:    option.Config.HubbleMetricsServer,
+		Handler: nil,
+	}
+	errChan := make(chan error, 1)
+
+	go func() {
+		if metricsTLSConfig != nil {
+			srv.TLSConfig = metricsTLSConfig.ServerConfig(&tls.Config{ //nolint:gosec
+				MinVersion: serveroption.MinTLSVersion,
+			})
+			errChan <- srv.ListenAndServeTLS("", "")
+		} else {
+			errChan <- srv.ListenAndServe()
+		}
+	}()
+	go func() {
+		err := <-errChan
+		if err != nil && !errors.Is(err, http.ErrServerClosed) {
+			log.WithError(err).Error("Unable to start Hubble metrics server")
+		}
+	}()
+
+	InitMetricsServerHandler(&srv, Registry, option.Config.EnableHubbleOpenMetrics)
+	err := InitMetrics(Registry, api.ParseMetricList(option.Config.HubbleMetrics), grpcMetrics)
+	if err != nil {
+		log.WithError(err).Error("Unable to setup metrics: %w", err)
+		return
+	}
+}
```

Comment:
All of this needs to happen before you launch the Go routines.

Reply:
Moved it up! Is it necessary, though? I'm thinking, in the test code `SetUpTestMetricsServer` starts the server with a `nil` handler, before calling `InitMetricsServerHandler`, and it passes. Are you worried about a prod scenario where the `launchHubble` thread  is pre-empted and a request reaches the running server before before the MUX is set up? 

(I may have earlier misread a StackOverflow post which said we couldn't swap handlers within a running mux, but in this PR the test swaps out the mux ref itself.)